### PR TITLE
delete unused ExprType

### PR DESCRIPTION
**Pull request by birdstorm**
Since tipb has deleted most of `ExprType`, now master breaks compilation.

Delete unused `ExprType` usage in our project.

### CODE REVIEW

**Comment by birdstorm** [comment]
PTAL @ilovesoup @Novemser

File: tikv-client/src/main/java/com/pingcap/tikv/types/YearType.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ protected void encodeProto(CodecDataOutput cdo, Object value) {
 
   @Override
   public ExprType getProtoExprType() {
-    return ExprType.Year;
+    return ExprType.Uint64;
```
Comment (Novemser) [inline tikv-client/src/main/java/com/pingcap/tikv/types/YearType.java]: Why we need to change this to `UInt64`?

Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/types/YearType.java]: year no longer exists in ExprType. Uint64 is its replacement.

Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/types/YearType.java]: Year has changed back to Int64. ICYMI @Novemser

**Pull request merged**